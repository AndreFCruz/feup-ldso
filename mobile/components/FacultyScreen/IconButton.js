import React from 'react';
import {View, Card, Icon, Button, Text} from 'native-base';
import specificStyles from '../../constants/SpecificStyles';

class IconButton extends React.Component {
    render() {
        return (
            <Card style={styles.card} transparent>
                <View style={styles.btnView}>
                    <Button style={[styles.linkBtn, specificStyles.mainMenuBtn.feup]} /*onPress={() => this.props.navigation.navigate('Videos', {name: name})}*/>
                        <Icon style={[styles.icon,specificStyles.mainMenuIcon.feup]} type="FontAwesome" name="film" />
                    </Button>
                </View>
                <View style={styles.subtitle}>
                    <Text style={styles.labelText}>Videos</Text>
                </View>
            </Card>
        );
    }
}

export { IconButton };

const styles = {

    card: {
        padding: 30,
       alignItems: 'center'
    },

    btnView: {
        flex:1,
        flexDirection: 'column',
    },

    subtitle: {
        flex:1,
        flexDirection: 'column',
        alignItems: 'center',
        marginTop:83,
    },


    labelText: {
        fontSize: 22
    },

    icon: {
        transform: [{
            rotate: '315deg'
        }]
    },

    linkBtn: {
        width: 90,
        height: 90,
        justifyContent: 'center',
        borderWidth: 1.2,
        borderRadius: 6,
        backgroundColor: 'white',
        transform: [{
            rotate: '45deg'
        }]
  
    }
}