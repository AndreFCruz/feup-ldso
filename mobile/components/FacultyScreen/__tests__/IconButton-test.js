import React from 'react';
import IconButton  from '../IconButton';
import {Card, View, Button} from 'native-base';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const navigation = { navigate: jest.fn()};

function setup(name = true) {
    let wrapper;

    if(name == false) {
        wrapper = shallow(<IconButton icon='videos' label='Videos' action={() => navigation.navigate('Video', {name:'feup'})}/>,
            {
                disableLifecycleMethods: true
            });
    } else {
        wrapper = shallow(<IconButton name='feup' icon='videos' label='Videos' action={() => navigation.navigate('Video', {name:'feup'})}/>,
            {
                disableLifecycleMethods: true
            });
    }
    return wrapper;

}

describe('Rendering ', () => {
    it('renders fact with content received', () => {
        const wrapper = setup();
        expect(wrapper.dive()).toMatchSnapshot();
    });
    it('do not render', () => {
        const wrapper = setup(false);
        expect(wrapper.type()).toEqual(null);
    });

});

describe('Click', () => {
    it('check button click', () => {
        const wrapper = setup();
        const spy = jest.spyOn(navigation, 'navigate');
        wrapper.find(Card).dive().find(View).at(0).dive().find(Button).simulate('click');
        expect(spy).toBeCalledWith('Video', {name: 'feup'});

    });
});