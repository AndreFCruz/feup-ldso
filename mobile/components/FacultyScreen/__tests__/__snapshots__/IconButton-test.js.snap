// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering  renders fact with content received 1`] = `
<Card
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderBottomWidth": 0,
        "borderColor": "#ccc",
        "borderLeftWidth": 0,
        "borderRadius": 2,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "elevation": null,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      },
    ]
  }
  transparent={true}
>
  <Styled(ViewNB)
    style={
      Object {
        "flex": 100,
        "justifyContent": "space-around",
      }
    }
  >
    <Styled(Button)
      style={
        Array [
          Object {
            "borderRadius": 17,
            "borderWidth": 2.2,
            "elevation": 10,
            "height": 180,
            "justifyContent": "center",
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 180,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#9f2d20",
          },
        ]
      }
    >
      <SvgUri
        fill="#9f2d20"
        fillAll={true}
        height={86.5}
        source={1}
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "315deg",
              },
            ],
          }
        }
        width={86.5}
      />
    </Styled(Button)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    style={
      Array [
        Object {
          "flex": 32,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 33.5,
          "textAlign": "center",
        }
      }
    >
      Videos
    </Styled(Text)>
  </Styled(ViewNB)>
</Card>
`;
